### PR TITLE
fix(amazonq): /dev to support upload of nested dockerfiles

### PR DESCRIPTION
**Pull request by michaellandi**
[POST RE:INVENT]

## Problem

Nested `Dockerfile` files were not supported with https://github.com/aws/aws-toolkit-vscode/pull/6107

## Solution

Look at path basename when comparing file to list of well known files.

I manually tested adding nested `Dockerfile`s to various folders and ensured that pattern matched.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by avi-alpert — APPROVED** [review]

**Review by neilk-aws — APPROVED** [review]

File: packages/core/src/shared/filetypes.ts [hunk 1]
```diff
@@ -355,7 +355,7 @@ export const wellKnownCodeFiles = new Set(['Dockerfile', 'Dockerfile.build'])
 export function isCodeFile(filename: string): boolean {
```
Comment (justinmk3) [inline packages/core/src/shared/filetypes.ts:355]: please rename `wellKnownCodeFiles` to something like `codefileBaseNames`, so that it follows the existing pattern of `codefileExtensions`.

File: packages/core/src/shared/filetypes.ts [hunk 2]
```diff
@@ -355,7 +355,7 @@ export const wellKnownCodeFiles = new Set(['Dockerfile', 'Dockerfile.build'])
...
     if (codefileExtensions.has(path.extname(filename).toLowerCase())) {
         return true
-    } else if (wellKnownCodeFiles.has(filename)) {
+    } else if (wellKnownCodeFiles.has(path.basename(filename))) {
         return true
     } else {
         return false
```
Comment (justinmk3) [inline packages/core/src/shared/filetypes.ts:361]: would the lowercase step also be needed?

please also revert the change that you made in the previous pr, so that this code is less verbose:

```typescript
    const ext = path.extname(filename).toLowerCase()
    return codefileExtensions.has(ext) || wellKnownCodeFiles.has(path.basename(filename)
```

File: packages/core/src/shared/filetypes.ts (hunk 1 above)
Comment (michaellandi) [reply packages/core/src/shared/filetypes.ts:355]: This already exists here and in the JetBrains extension: https://github.com/aws/aws-toolkit-jetbrains/blob/07ccd110d741c4bf591bfcbc5db86a2a7022313f/plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt#L72

File: packages/core/src/shared/filetypes.ts (hunk 2 above)
Comment (michaellandi) [reply packages/core/src/shared/filetypes.ts]: I do think we want a case sensitive match in this case.  `Dockerfile` is the convention when used without an extension: https://docs.docker.com/build/concepts/dockerfile/#filename.

File: packages/core/src/shared/filetypes.ts (hunk 1 above)
Comment (justinmk3) [reply packages/core/src/shared/filetypes.ts:355]: Please rename it.

File: packages/core/src/shared/filetypes.ts (hunk 2 above)
Comment (justinmk3) [reply packages/core/src/shared/filetypes.ts]: That's the case for `Dockerfile`, but likely not other "filenames without an extension" . I guess we can wait until we encounter such files.

File: packages/core/src/shared/filetypes.ts (hunk 1 above)
Comment (michaellandi) [reply packages/core/src/shared/filetypes.ts:355]: Okay, done.
